pages/item/[id].js

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import initialItems from "@/public/assets/shopping-items.json";
+import initialCategories from "@/public/assets/categories.json";
+import Link from "next/link";
+
+const StyledItemDetails = styled.div`
+  max-width: 600px;
+  margin: 0 auto;
+  padding: 20px;
+
+  img {
+    max-width: 100%;
+    height: auto;
+    border-radius: 5px;
+  }
+
+  .back-link {
+    display: inline-block;
+    margin-bottom: 20px;
+    text-decoration: none;
+    color: #007bff;
+    &:hover {
+      text-decoration: underline;
+    }
+  }
+`;
+
+const ItemDetails = () => {
+  const router = useRouter();
+  const { id } = router.query;
+  console.log(id);
+  const item = initialItems.find((item) => item.id === id);
+
+  if (!item) {
+    return <div>Item not found</div>;
+  }
+
+  return (
+    <StyledItemDetails>
+      <Link className="back-link" href="/">
+        ← Back to Shopping List
+      </Link>
```

Comment:
You are currently styling the Link using a nested class.

You can extend the Link component and apply styles to it:

```js
const StyledLink = styled(Link)`
display: inline-block;
    margin-bottom: 20px;
    text-decoration: none;
    color: #007bff;
    &:hover {
      text-decoration: underline;
    }
`
```

This is the approach I would prefer, since you would not mix CSS with styled-components by doing so